ingest/indexer/service/blockprocessor/block_updates_indexer_block_process_strategy.go

```diff
@@ -0,0 +1,49 @@
+package blockprocessor
+
+import (
+	"github.com/cosmos/cosmos-sdk/types"
+
+	commondomain "github.com/osmosis-labs/osmosis/v25/ingest/common/domain"
+	"github.com/osmosis-labs/osmosis/v25/ingest/indexer/domain"
+)
+
+type blockUpdatesIndexerBlockProcessStrategy struct {
+	client            domain.Publisher
+	poolExtractor     commondomain.PoolExtractor
+	poolPairPublisher domain.PairPublisher
+}
```

Comment:
Note :tracking tests for this component here: https://linear.app/osmosis/issue/IND-72/unit-test-block-updates-indexer-block-process-strategygo